dex/farm-with-locked-rewards/src/lib.rs

```diff
@@ -237,7 +237,13 @@ pub trait Farm:
             );
         }
 
+        let mut storage_cache = StorageCache::new(self);
+        NoMintWrapper::<Self>::generate_aggregated_rewards(self, &mut storage_cache);
+
         let boosted_rewards = self.claim_only_boosted_payment(user);
+
+        self.set_farm_supply_for_current_week(&storage_cache.farm_token_supply);
```

Comment:
do we have to do this every single time ?

Reply:
Commented above. Also, this is how it already works for the other endpoints (enter, exit, claim).